src/data/schema.rs

```diff
@@ -22,13 +31,55 @@ impl ColumnDefExt for ColumnDef {
             .any(|ColumnOptionDef { option, .. }| option == &ColumnOption::Null)
     }
 
+    #[cfg(feature = "auto-increment")]
+    fn is_auto_incremented(&self) -> bool {
+        self.options
+            .iter()
+            .any(|ColumnOptionDef { option, .. }| option.is_auto_increment())
+    }
+
     fn get_default(&self) -> Option<&Expr> {
         self.options
             .iter()
-            .filter_map(|ColumnOptionDef { option, .. }| match option {
+            .find_map(|ColumnOptionDef { option, .. }| match option {
                 ColumnOption::Default(expr) => Some(expr),
                 _ => None,
             })
-            .next()
+    }
+}
+
+pub trait ColumnOptionExt {
+    fn is_auto_increment(&self) -> bool;
+}
+
+#[cfg(not(feature = "auto-increment"))]
+impl ColumnOptionExt for ColumnOption {
+    fn is_auto_increment(&self) -> bool {
+        false
+    }
+}
+
+#[cfg(feature = "auto-increment")]
+impl ColumnOptionExt for ColumnOption {
+    fn is_auto_increment(&self) -> bool {
+        matches!(self,
+        ColumnOption::DialectSpecific(tokens)
+            if matches!(
+                tokens[..],
+                [
+                    Token::Word(Word {
+                        keyword: Keyword::AUTO_INCREMENT,
+                        ..
+                    }),
+                    ..
+                ] | [
+                    Token::Word(Word {
+                        keyword: Keyword::AUTOINCREMENT,
+                        ..
+                    }),
+                    ..
+                ]
```

Comment:
Which one? I think you made a mistake with your comment.

If we were to support one I would think that `AUTO_INCREMENT` would be better.

I added them both for ease of use, they mean the same thing and it would be easier to use for more people.
Saying that, however, I'm indifferent to whether we keep it or not, happy to change it to just one.

Reply:
Neither `AUTOINCREMENT` and `AUTO_INCREMENT` is ANSI SQL, so it's all up to us to decide whatever to use.

I thought it would be better to keep only one, because in anyway when we need to explain this optional auto increment feature, it's simple to say using a single keyword.
e.g. MySQL -> `AUTO_INCREMENT`,  SQLite -> `AUTOINCREMENT`

I slightly prefer to use `AUTOINCREMENT` because I'm not familiar with using `_` in `CREATE TABLE` syntax.
It is usually more common to simply use `PRIMARY KEY`, `NOT NULL` -> spaces rather than `_`.
e.g. also in PostgreSQL syntax, there no `_` use. - https://www.postgresql.org/docs/9.1/sql-createtable.html

However, MySQL is too famous and it uses `_` to `AUTO_INCREMENT` so I'm ok with choosing either `AUTO_INCREMENT` or `AUTOINCREMENT`.

---
ok. let's take only one, either `AUTO_INCREMENT` or `AUTOINCREMENT`.
For what to choose, you choose and I'll follow.

Reply:
I'll make it `AUTO_INCREMENT` for now then, we can always change it later :-)

Reply:
I also like `AUTO_INCREMENT`, at least I like the MySQL's auto increment approach than SQLite's one. MySQL's one is more simple, straightforward and easy to implement, I think.
sounds good :)